### ISSUE TITLE
6.2 release general checklist

### ISSUE THREAD

**Issue by begriffs** [citusdata/citus_docs#366]
- [x] s/6\\.1/6.2/g
- [ ] Verify / Update major-version upgrade steps (Joe)
    - ❌ 5.2.2->6.2 fails. To reproduce: https://gist.github.com/begriffs/935b97623d52c691246840e1d3e042c4
- [x] Verify / Update minor-version upgrade steps (Joe)
    - From 6.1-17 to 6.2-4
- [x] Check / Document new UDF's added in 6.2
- [x] Check / Document new config options added in 6.2
- [x] Update links from Readme and Citus web site to newer version of docs if required
- [x] Update links to new CloudFormation template (Sumedh)
- [x] Verify Installation steps with new packages (6.2-4)
    - Ubuntu 16.04
    - Ubuntu 14.04
    - Debian Jessie
    - Debian Wheezy
    - Amazon Linux 2017.03.0
    - CentOS 7.3
    - CentOS 6.9
    - RHEL 7.2
    - RHEL 6.5
    - ❌ Fedora 25 **No package citus62_96 available.**
    - Fedora 24
- [ ] Update all installation steps to use new package name (CloudFormation, Website instructions) (Sumedh)
- [x] Create new 6.2 release branch and set as default ver in RTD (last step)

**Comment by begriffs** [comment]
@sumedhpathak here's a list of UDFs that are either not in the most recent docs at all, or are there but do not have entries in the reference section.

```sql
DROP FUNCTION master_drop_sequences(text[], text, bigint);
CREATE FUNCTION master_drop_sequences(sequence_names text[])

CREATE FUNCTION upgrade_to_reference_table(table_name regclass)


CREATE FUNCTION master_disable_node(nodename text, nodeport integer)

CREATE FUNCTION master_dist_local_group_cache_invalidate()
    IS 'register node cache invalidation for changed rows';


CREATE FUNCTION worker_apply_sequence_command(text)
    IS 'create a sequence which products globally unique values';

CREATE FUNCTION isolate_tenant_to_new_shard(table_name regclass, tenant_id "any", cascade_option text DEFAULT '')

CREATE FUNCTION stop_metadata_sync_to_node(nodename text, nodeport integer)

CREATE FUNCTION column_to_column_name(table_name regclass, column_var_text text)
    IS 'convert the textual Var representation to a column name';

CREATE FUNCTION lock_shard_resources(lock_mode int, shard_id bigint[])
    IS 'lock shard resource to serialise non-commutative writes';

CREATE FUNCTION lock_shard_metadata(lock_mode int, shard_id bigint[])
    IS 'lock shard metadata to prevent writes during metadata changes';

CREATE FUNCTION master_drop_distributed_table_metadata(logicalrelid regclass,
                                          			   schema_name text,
                                          			   table_name text)
```

Should I create reference sections for all of them, or only for those which have no presence at all in docs?

**Comment by begriffs** [comment]
Also these are the GUCs that are not covered in docs:

```
citus.enable_deadlock_prevention
citus.enable_router_execution
citus.enable_unique_job_ids
citus.enable_version_checks
citus.expire_cached_shards
citus.explain_distributed_queries
citus.log_multi_join_order
citus.log_remote_commands
citus.multi_task_query_log_level
citus.node_connection_timeout
citus.replication_model
citus.shard_placement_policy
citus.subquery_pushdown
citus.version
citus.worker_list_file
```

Of the previous list, these are the ones that have changed since tag v6.1.0:

```
citus.multi_task_query_log_level
citus.version
citus.enable_version_checks
citus.enable_unique_job_ids
```

Which ones should I document?

**Comment by sumedhpathak** [comment]
```
DROP FUNCTION master_drop_sequences(text[], text, bigint);
CREATE FUNCTION master_drop_sequences(sequence_names text[])
```
These are part of the DROP trigger. They are not meant to be called manually. We shouldn't document them.

```
CREATE FUNCTION upgrade_to_reference_table(table_name regclass)
```
We documented this here: https://docs.citusdata.com/en/v6.1/dist_tables/ddl.html#reference-tables. If we don't have a reference section for it, happy to add it.

```
CREATE FUNCTION master_disable_node(nodename text, nodeport integer)
```
We have an issue for it already: https://github.com/citusdata/citus_docs/issues/224. Let's discuss this during prioritization.

```
CREATE FUNCTION master_dist_local_group_cache_invalidate()
    IS 'register node cache invalidation for changed rows';
```
Appears to be an internal function.

```
CREATE FUNCTION worker_apply_sequence_command(text)
    IS 'create a sequence which products globally unique values';
```
Used in MX apparently, and still seems for internal use. Skip.

```
CREATE FUNCTION isolate_tenant_to_new_shard(table_name regclass, tenant_id "any", cascade_option text DEFAULT '')
```
Didn't we document Tenant Isolation? If we don't have this in reference, let's add this and link to the usage section.

```
CREATE FUNCTION stop_metadata_sync_to_node(nodename text, nodeport integer)
```
MX function. Not for public use. Skip.

```
CREATE FUNCTION column_to_column_name(table_name regclass, column_var_text text)
    IS 'convert the textual Var representation to a column name';
```
Again, didn't we recently document this? If not in reference, happy to add it, and link to usage.

```
CREATE FUNCTION lock_shard_resources(lock_mode int, shard_id bigint[])
    IS 'lock shard resource to serialise non-commutative writes';

CREATE FUNCTION lock_shard_metadata(lock_mode int, shard_id bigint[])
    IS 'lock shard metadata to prevent writes during metadata changes';

CREATE FUNCTION master_drop_distributed_table_metadata(logicalrelid regclass,
                                          			   schema_name text,
                                          			   table_name text)
```
Rest of the above all seem internal use.

**Comment by sumedhpathak** [comment]
`citus.enable_deadlock_prevention:`
Do not document. NSFW. :)

`citus.enable_router_execution`
Primarily for debugging etc, allows turning off router execution, and forces through real-time. Skip.

`citus.enable_unique_job_ids`
Text from the GUC definition: "This should usually be enabled, but can be disabled for repeatable output in regression tests." Not for external use, so skip.

`citus.enable_version_checks`
Setting to false skips erroring out for version upgrades. Not something we want to document unless there is a specific use-case. Skip.

`citus.expire_cached_shards`
Skip. This may change as we now have reference tables.

`citus.explain_distributed_queries`
Is default to true, and turned off only for regression tests. Skip documentation.

`citus.log_multi_join_order`
For dev debugging, skip documentation.

`citus.log_remote_commands`
Skip documentation (for now). Debugging purposes, but might be useful in the future.

`citus.multi_task_query_log_level`
Allows one to error or warn on queries which run in parallel (for multi-tenant use-cases). Probably worth documenting as part of a migration step. (e.g. set to ERROR during testing to see if all queries have been migrated to suitable multi-tenant ones). @begriffs Open up an issue for this?

`citus.node_connection_timeout`
Skip (for now), but might be useful in the future. Add to new issue?

`citus.replication_model`
We've discussed this one, and have an issue on it: https://github.com/citusdata/citus_docs/issues/271

`citus.shard_placement_policy`
Let's open up an issue for this one. This hasn't come up frequently as a customer facing issue, but lets create an issue for it.

`citus.subquery_pushdown`
Skip.

`citus.version`
Happy to document this one. This seems to only show the current version, and isn't settable. From Postgres: "INTERNAL options cannot be set by the user at all, but only through
 * internal processes ("server_version" is an example).  These are GUC
 * variables only so they can be shown by SHOW, etc." @begriffs Open up an issue? Or just document.

`citus.worker_list_file`
Skip. Deprecated.

**Comment by begriffs** [comment]
Ran into a problem when testing a major version upgrade.

1. Installed Citus 5.2-1 on postgres 9.5.7 on Ubuntu 16.04 with a coordinator on 5432 and worker on 5433
1. Successfully created schema and distributed data
1. Built citus from master and did `sudo make install`
1. Attempted to restart db in order to then run alter extension
1. Unable to restart, `journalctl -xe` says:
   ```
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]: 2017-05-17 18:03:59.950 UTC [16648] postgres@postgres ERROR:  loaded Citus library version differs from installed extension version
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]: 2017-05-17 18:03:59.950 UTC [16648] postgres@postgres DETAIL:  Loaded library requires 6.2, but the installed extension version is 5.2-1.
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]: 2017-05-17 18:03:59.950 UTC [16648] postgres@postgres HINT:  Run ALTER EXTENSION citus UPDATE and try again.
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]: 2017-05-17 18:03:59.950 UTC [16648] postgres@postgres STATEMENT:  SELECT d.datname as "Name",
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:                pg_catalog.pg_get_userbyid(d.datdba) as "Owner",
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:                pg_catalog.pg_encoding_to_char(d.encoding) as "Encoding",
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:                d.datcollate as "Collate",
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:                d.datctype as "Ctype",
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:                pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:         FROM pg_catalog.pg_database d
   May 17 18:03:59 ip-172-30-2-64 postgresql@9.5-worker[16600]:         ORDER BY 1;
   May 17 18:03:59 ip-172-30-2-64 systemd[1]: Failed to start PostgreSQL Cluster 9.5-worker.
```

**Comment by begriffs** [comment]
6.2 docs are released, so closing this issue.
